packages/plugin-phone/src/call.js

```diff
@@ -832,22 +833,28 @@ const Call = SparkPlugin.extend({
         offerToReceiveAudio: recvOnly || !!options.constraints.audio,
         offerToReceiveVideo: recvOnly || !!options.constraints.video
       });
+    }
 
-      this.media.set({
+    const mediaOptions = {};
+    if (options.constraints) {
+      mediaOptions.constraints = {
         audio: options.constraints.audio,
         video: options.constraints.video
-      });
+      };
     }
 
     if (options.offerOptions) {
-      if (options.offerOptions.offerToReceiveAudio) {
-        this.media.set({offerToReceiveAudio: options.offerOptions.offerToReceiveAudio});
-      }
-      if (options.offerOptions.offerToReceiveVideo) {
-        this.media.set({offerToReceiveVideo: options.offerOptions.offerToReceiveVideo});
-      }
+      mediaOptions.offerOptions = {
+        offerToReceiveAudio: options.offerOptions.offerToReceiveAudio,
+        offerToReceiveVideo: options.offerOptions.offerToReceiveVideo
+      };
     }
 
+    if (mediaOptions.offerOptions || mediaOptions.constraints) {
```

Comment:
you probably don't need this conditional; set should set it as an empty object and do nothing, but this might be a theoretical perf win.